ssl/record/ssl3_record.c

```diff
@@ -1481,7 +1478,11 @@ int tls1_mac(SSL *ssl, SSL3_RECORD *rec, unsigned char *md, int sending)
         BIO_printf(trc_out, "md:\n");
         BIO_dump_indent(trc_out, md, md_size, 4);
     } OSSL_TRACE_END(TLS);
-    return 1;
+    ret = 1;
+ end:
+    if (!stream_mac)
```

Comment:
There is no need for this `if` as hmac will be NULL otherwise and the EVP_MD_CTX_free() handles NULL gracefully.